### PR TITLE
Allow debugging cleanbuild runs.

### PR DESCRIPTION
**Pull request by sergiusens**
When using --debug a shell is spawned into the cleanbuild run
to allow debugging and troubleshooting.

LP: #1609803
    
Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by josepht** [comment]
+1

File: snapcraft/internal/lxd.py [hunk 1]
```diff
@@ -67,7 +67,11 @@ def _create_container(self):
                 self._container_name])
             yield
         finally:
-            check_call(['lxc', 'stop', self._container_name])
+            # We do this as lxc stop does not print anything out as
+            # it does when creating the container and stopping can take
+            # a bit.
```
Comment (elopio) [inline snapcraft/internal/lxd.py]: This text is confusing. Maybe just: # Stopping takes a while and lxc doesn't print anything.

File: snapcraft/internal/lxd.py [hunk 2]
```diff
@@ -76,9 +80,17 @@ def execute(self):
...
-            self._container_run(
-                ['snapcraft', 'snap', '--output', self._snap_output])
-            self._pull_snap()
+            try:
+                self._container_run(
+                    ['snapcraft', 'snap', '--output', self._snap_output])
+            except CalledProcessError as e:
+                if self._project_options.debug:
+                    logger.info('Debug mode enabled, dropping into a shell')
+                    self._container_run(['bash', '-i'])
```
Comment (elopio) [inline snapcraft/internal/lxd.py]: nice!

File: snapcraft/main.py [hunk 3]
```diff
@@ -48,7 +48,8 @@
   -h --help                             show this help message and exit
   -v --version                          show program version and exit
   -d --debug                            print debug information while executing
-                                        (including backtraces)
+                                        (including backtraces). When calling
+                                        cleanbuild opens a shell on failures
```
Comment (elopio) [inline snapcraft/main.py]: I would be a little more verbose. When used with cleanbuild, it opens a shell in case of failure.

**Comment by elopio** [comment]
:+1: all good. What about a manual test for the actual cleanbuild execution?

**Comment by sergiusens** [comment]
@elopio done

**Comment by elopio** [comment]
:shipit:

**Pull request merged**